packages/core/src/App/Components/Routes/binary-routes.jsx

```diff
@@ -4,23 +4,34 @@ import { Loading } from '@deriv/components';
 import getRoutesConfig from 'App/Constants/routes-config';
 import RouteWithSubRoutes from './route-with-sub-routes.jsx';
 import { observer, useStore } from '@deriv/stores';
-import { getPositionsV2TabIndexFromURL, isDTraderV2, routes } from '@deriv/shared';
+import { getPositionsV2TabIndexFromURL, isDTraderV2, isDTraderV2Width, routes } from '@deriv/shared';
+import { useGrowthbookGetFeatureValue } from '@deriv/hooks';
 
 const BinaryRoutes = observer(props => {
     const { ui, gtm } = useStore();
     const { promptFn, prompt_when } = ui;
     const { pushDataLayer } = gtm;
     const location = useLocation();
-    const is_dtrader_v2 =
-        isDTraderV2() && (location.pathname.startsWith(routes.trade) || location.pathname.startsWith('/contract/'));
+    const [dtrader_v2_enabled_gb] = useGrowthbookGetFeatureValue({
+        featureFlag: 'dtrader_v2_enabled',
+        defaultValue: false,
+    });
+    const [dtrader_v2_enabled, setDTraderV2Enabled] = React.useState(false);
+    React.useEffect(() => {
+        setDTraderV2Enabled(
```

Comment:
so really, we need to wait for GB to be loaded (and I hope GB is smart enough to cache the response for subsequent calls)

Reply:
@wojciech-deriv A flicker is expected for now because for a given feature flag growthbook always returns false and then true. The hook that I have created is not accurate but for now we don't have much option. Prince is working on Analytics package to give us the status of growthbook (eg:- error conditions). Once that is implemented I will come back and back the change. Also ensured that flicker will only appear in dtrader and that too only on v2